Sources/ArgumentParser/Usage/UsageGenerator.swift

```diff
@@ -339,16 +339,36 @@ extension ErrorMessageGenerator {
     }
   }
   
-  func unableToParseValueMessage(origin: InputOrigin, name: Name?, value: String, key: InputKey) -> String {
+  func unableToParseValueMessage(origin: InputOrigin, name: Name?, value: String, key: InputKey, error: Error?) -> String {
     let valueName = arguments(for: key).first?.valueName
-    switch (name, valueName) {
-    case let (n?, v?):
+    
+    // We want to make the "best effort" in producing a custom error message.
+    // We favour `LocalizedError.errorDescription` and fall back to
+    // CustomStringConvertible. To opt-in return your custom error mesage
+    // in the `description` property of CustomStringConvertible.
+    let customErrorMessage: String? = {
+        switch error {
+            case is LocalizedError where (error as? LocalizedError)?.errorDescription != nil:
+                return (error as? LocalizedError)?.errorDescription
+            case let err?:
+                return String(describing: err)
+            default:
+                return nil
+        }
+    }()
+    
+    switch (name, valueName, customErrorMessage) {
```

Comment:
Since we should be doing the same thing in each case, this switch statement would be easier to understand if we just concatenate the `"The value ..."` text with the custom error message (if it exists).

Reply:
Trying to be a bit too clever and adding complexity in the process. 

I think some of these cases will never have custom messages, but it will just work if they ever do start passing in errors.